tasksCode/AnyType.hpp

```diff
@@ -0,0 +1,412 @@
+	~AnyType() {
+		delete valueData;
+	}
+};
+
+// Set the default internal value' data
+void AnyType::SetValueData()
+{
+	if (valueData == nullptr)
+		valueData = new ValueData;
+	// Set all valueData bytes to 0 (the long long int has 64 bytes
+	// which is the max fundamental datatype size)
+	(*valueData).valueLLInt = 0;
+}
+
+// Set the internal value' data, based on the give value
+template <typename Type>
+void AnyType::SetValueData(const Type & value)
+{
+	if (valueData == nullptr)
+		valueData = new ValueData;
+	// Iterate through possible types to define what value interpretation should be choosen
+	// The write the value into the value' data according to the correct interpreation
+	auto valueTypeId = typeid(value).name();
+	if		(valueTypeId == typeid( bool					).name()) (*valueData).valueBool = static_cast<bool>(value);
+	else if (valueTypeId == typeid( char					).name()) (*valueData).valueChar = static_cast<char>(value);
+	else if (valueTypeId == typeid( unsigned char			).name()) (*valueData).valueUChar = static_cast<unsigned char>(value);
+	else if (valueTypeId == typeid( int						).name()) (*valueData).valueLLInt = static_cast<long long>(value);
+	else if (valueTypeId == typeid( short int				).name()) (*valueData).valueLLInt = static_cast<long long>(value);
+	else if (valueTypeId == typeid( long int				).name()) (*valueData).valueLLInt = static_cast<long long>(value);
+	else if (valueTypeId == typeid( long long int			).name()) (*valueData).valueLLInt = static_cast<long long>(value);
+	else if (valueTypeId == typeid( unsigned int			).name()) (*valueData).valueLLUInt = static_cast<unsigned long long>(value);
+	else if (valueTypeId == typeid( unsigned short int		).name()) (*valueData).valueLLUInt = static_cast<unsigned long long>(value);
+	else if (valueTypeId == typeid( unsigned long int		).name()) (*valueData).valueLLUInt = static_cast<unsigned long long>(value);
+	else if (valueTypeId == typeid( unsigned long long int	).name()) (*valueData).valueLLUInt = static_cast<unsigned long long>(value);
+	else if (valueTypeId == typeid( float					).name()) (*valueData).valueLDouble = static_cast<long double>(value);
+	else if (valueTypeId == typeid( double					).name()) (*valueData).valueLDouble = static_cast<long double>(value);
+	else if (valueTypeId == typeid( long double				).name()) (*valueData).valueLDouble = static_cast<long double>(value);
+	else throw std::bad_typeid{ }; // Cast exception if the type is unknown
+}
+
+// Set the default internal value' type
+void AnyType::SetValueType()
+{
+	valueType = ValueType::TypeVoid;
+}
+
+// Set the internal value' type, based on the give value
+template <typename Type>
+void AnyType::SetValueType(const Type & value)
+{
+	// Iterate through possible types and define what type does the value have
+	// Then write the type to the state variable valueType
+	auto valueTypeId = typeid(value).name();
+	if		(valueTypeId == typeid( bool					).name()) valueType = ValueType::TypeBoolean;
```

Comment:
I would suggest to use GetValueType function (line 367) and switch construction

Reply:
GetValueType allows get the type of the internal value, but not the type of parameter. Moreover its return value is string with the name of the internal type.
In contrast here we have to define the type of the given parameter. Since this function is template, the only way to define the type of parameter is to use typeid

Reply:
You can`t use GetValueType() to this function as @PeterLysyk said, but using swich construction instead of if else constuction is recomended ^_^

Reply:
Compiler refuse to compile **switch case** statements with **typeinfo(Type)** or **typeinfo(Type).name()** types. Instead, **switch case** is applicable only to integer-like types or enumerations.